### PR TITLE
Fix for stage3 when setting different communication data type

### PR DESCRIPTION
**Pull request by BacharL**

### CODE REVIEW

File: deepspeed/runtime/zero/stage3.py [hunk 1]
```diff
@@ -1158,8 +1158,7 @@ def __reduce_and_partition_ipg_grads(self, safe_mode: bool = False) -> None:
 
     @instrument_w_nvtx
     def __avg_scatter_contiguous_grads(self, buffer_to_reduce: Tensor) -> List[Tensor]:
-        dtype = buffer_to_reduce.dtype
-        if self.communication_data_type == self.dtype:
+        if self.communication_data_type != self.dtype:
```
Comment (tjruwase) [inline deepspeed/runtime/zero/stage3.py:1161]:
```suggestion
        dtype = buffer_to_reduce.dtype
        if self.communication_data_type != dtype:
```

File: deepspeed/runtime/zero/stage3.py [hunk 2]
```diff
@@ -1159,7 +1159,7 @@ def __reduce_and_partition_ipg_grads(self, safe_mode: bool = False) -> None:
     @instrument_w_nvtx
     def __avg_scatter_contiguous_grads(self, buffer_to_reduce: Tensor) -> List[Tensor]:
         dtype = buffer_to_reduce.dtype
-        if self.communication_data_type == self.dtype:
+        if self.communication_data_type != dtype:
```
Comment (tjruwase) [inline deepspeed/runtime/zero/stage3.py:1162]:
```suggestion
        if self.communication_data_type != self.dtype:
```

Comment (tjruwase) [reply deepspeed/runtime/zero/stage3.py:1162]: I think line 1161 is redundant and could be deleted. What do you think?

Comment (hablb) [reply deepspeed/runtime/zero/stage3.py:1162]: I agree. Thanks.

**Review by tjruwase — APPROVED** [review]

File: deepspeed/runtime/zero/stage3.py (hunk 1 above)
Comment (hablb) [reply deepspeed/runtime/zero/stage3.py:1217]: I agree.

**Pull request merged**